### PR TITLE
HAAR-105 add new prison

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: integration_tests/integration/prison-register-add.spec.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+context('Prison register - Add new prison', () => {
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubLogin')
+    cy.task('stubAuthUser')
+    cy.task('stubGetPrisonsWithFilter', [albanyPrison, { ...belmarshPrison }, moorlandPrison])
+    cy.task('stubGetPrison', albanyPrison)
+    cy.task('stubGetPrison', moorlandPrison)
+    cy.task('stubAddPrison')
+    // cy.task('stubAddPrisonAddress')
```
Comment (AlTurner-MOJ) [inline integration_tests/integration/prison-register-add.spec.ts:18]: remove this if not needed

File: server/routes/prisonRegister/prisonRegisterController.test.ts [hunk 2]
```diff
@@ -614,38 +613,5 @@ describe('Prison Register controller', () => {
         })
       })
     })
-
```
Comment (AlTurner-MOJ) [inline server/routes/prisonRegister/prisonRegisterController.test.ts:617]: Did you mean to remove this block?

File: server/views/pages/prison-register/addNewPrisonAddress.njk [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+            classes: "govuk-fieldset__legend--l",
+            isPageHeading: true
+          }
+        }) %}
+
+            {{ govukDetails({
+              summaryText: "What should I do if this prison has multiple addresss at different addresses?",
+              text: "Many services are only interested in the main address for a prison and they would typically only
+               show their users the prison and main address. However, where the physical address is significant,
+                for instance the location where a prisoner needs to be sent to, then multiple addresss should be
```
[the remark this answers is not in the document]
Comment (karenmillermoj) [reply server/views/pages/prison-register/addNewPrisonAddress.njk]: fixed

File: integration_tests/integration/prison-register-add.spec.ts (hunk 1 above)
Comment (karenmillermoj) [reply integration_tests/integration/prison-register-add.spec.ts]: removed

File: server/routes/prisonRegister/addNewPrisonAddressValidator.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { Request } from 'express'
+import type { AddNewPrisonForm } from 'prisonForms'
+import { validate as validateSync } from '../../validation/validation'
+
+export default function validate(form: AddNewPrisonForm, req: Request): string {
```
[the remark this answers is not in the document]
Comment (karenmillermoj) [reply server/routes/prisonRegister/addNewPrisonAddressValidator.ts:5]: I did - I think this needs doing but I had already started using the "other" way....and I already had lots of files to change.  Would be good to refactor to use that code (possibly in another PR)

File: server/routes/prisonRegister/prisonRegisterController.test.ts (hunk 2 above)
Comment (karenmillermoj) [reply server/routes/prisonRegister/prisonRegisterController.test.ts:617]: added back in

**Review by AlTurner-MOJ — APPROVED** [review]

File: server/routes/prisonRegister/prisonRegisterController.test.ts [hunk 5]
```diff
@@ -614,38 +613,5 @@ describe('Prison Register controller', () => {
...
-    describe('submitAmendPrisonAddress', () => {
```
[the remark this answers is not in the document]
Comment (karenmillermoj) [reply server/routes/prisonRegister/prisonRegisterController.test.ts]: will address in separate PR

**Pull request merged**